bindings/ruby/lib/fdbimpl.rb

```diff
@@ -843,7 +843,7 @@ def get_range_start_with(prefix, options={}, &block)
     end
 
     def get_estimated_range_size_bytes(begin_key, end_key)
-      if begin_key.nil? || end_key.nil?
+      if FDB.get_api_version()>= 700 && (begin_key.nil? || end_key.nil?)
```

Comment:
Looking at this function before the change, it already would produce an error if you passed `nil` for either begin or end. You could still choose to detect this and report an error if you want, but interestingly I think the error being added here in some ways has less information than the existing one (which would indicate that the error was a result of using `nil`).

I'll also point out that it doesn't appear any other functions have special validation for `nil` keys.

I think, then, that you could just undo this change altogether, but if you wanted to keep it, it's probably not necessary to version it.

Reply:
The nil check of this function was added in commit acdf04ed187b1b894179387d8843cfb7e46a8f85, which is part of this PR. I added the versioning code later. But I think you are right there is no nil check done in other functions.

Reply:
If you don't explicitly check for `nil`, you'll still get an error for trying to use the `nil` a few lines later:

```
Traceback (most recent call last):
        3: from /usr/bin/irb:11:in `<main>'
        2: from (irb):7
        1: from /media/psf/work/code/foundationdb/bindings/ruby/lib/fdbimpl.rb:825:in `get_estimated_range_size_bytes'
NoMethodError (undefined method `bytesize' for nil:NilClass)
```

This error is indicating that something was unexpectedly `nil` which is in some ways a more specific error than the newly added one.

Reply:
Ah good point. It is indeed a better error message. Let me undo my changes.